python/cudf/cudf/__init__.py

```diff
@@ -96,6 +97,7 @@
 
     _setup_numba_linker(_PTX_FILE)
 
+    patch_numba_codegen_if_needed()
```

Comment:
This hack will go away

Reply:
I think it can be removed from this PR itself. Now that we've verified numpy 1.24 support, I would recommend removing the numba-related changes in this PR that you're using in order to allow running numba 0.57 (which is necessary to use numpy 1.24). We'll still get it tested because our CUDA 12 wheel builds will patch us to use 0.57 anyway (but with CUDA 12 we don't use cubinlinker/ptxcompiler so we don't need any edits for those). Then when we bump our numba to 0.57 tests should pass thanks to this PR.

Reply:
I have one other question on this PR, would wait to make changes here until everything else is resolved in case you need to run more tests locally.